prow/config/config.go

```diff
@@ -144,12 +145,33 @@ type ProwConfig struct {
 	DefaultJobTimeout *metav1.Duration `json:"default_job_timeout,omitempty"`
 }
 
+type InRepoConfig struct {
+	// Enabled describes whether InRepoConfig is enabled for a given repository. This can
+	// be set globally, per org or per repo using '*', 'org' or 'org/repo' as key. The
+	// narrowest match always takes precedence.
+	Enabled map[string]*bool
+}
+
 // InRepoConfigEnabled returns whether InRepoConfig is enabled. Currently
 // a no-op that always returns false, as the underlying feature is not implemented
 // yet. See https://github.com/kubernetes/test-infra/issues/13370 for a current
 // status.
-func (jc *JobConfig) InRepoConfigEnabled(identifier string) bool {
-	return jc.FakeInRepoConfig != nil
+func (c *Config) InRepoConfigEnabled(identifier string) bool {
+	// Used in tests
+	if c.FakeInRepoConfig != nil {
```

Comment:
Wait, did we not change this not to be a global set by tests? I think we need to do that ASAP -- this is a bad pattern and often when we have code handling test cases in production code we have bugs.

Reply:
Its a Property of `JobConfig`, initially it was a package-level var. I do agree that its not great to have branches for test usage in production code. How would you suggest to changes this?

Reply:
We would have to factor out the "get me my presubmits" functionality behind an interface and pass in something else that implements it to mock it

Reply:
ah okay, yeah that makes sense, I will create a follow-up for that